lib/spack/spack/test/cmd/dev_build.py

```diff
@@ -215,6 +215,45 @@ def test_dev_build_env(tmpdir, install_mockery, mutable_mock_env_path):
         assert f.read() == spec.package.replacement_string
 
 
+def test_dev_build_env_with_vars(tmpdir, install_mockery, mutable_mock_env_path, monkeypatch):
+    """Test Spack does dev builds for packages in develop section of env (path with variables)."""
+    # setup dev-build-test-install package for dev build
+    build_dir = tmpdir.mkdir("build")
+    spec = spack.spec.Spec("dev-build-test-install@0.0.0 dev_path=%s" % build_dir)
```

Comment:
Can you grab it from the concrete env instead of concretizing twice? (Although TIL you can set `dev_path` before concretizing)

Reply:
Not sure I get what you mean, probably I'm missing something.

The environment at this point is yet to be created, and this is just a concretization of the package in the wild so that it can be later used for comparison.

I can try reversing the logic of the test, to see if we can create the environment first and then get the package information from within the environment. To me it looks like the test semantic changes a bit. Let me know what you think and in case I can have a look (FIY I just copy-pasted the test above this).

Reply:
I guess it's fine like this, but in principle it could also just concretize the env and check the `dev_path` variant, and maybe stage dir.